### DISCUSSION TITLE
how do u use vm.runtime.startHats

### DISCUSSION THREAD

**Discussion by dumorando** [PenguinMod/PenguinMod-Home#99] [Q&A] [answered]
```js
Scratch.vm.runtime.startHats('dumoUI_event' + html, {
    ID: 'test'
}, Scratch.vm.runtime.getSpriteTargetByName(util.thread.target.getName()));
```
it is not running the hats with the id argument set to test
it just says  s[e] is undefined
please hlep

**Comment by RedMan13** [comment]
> ```js
> Scratch.vm.runtime.startHats('dumoUI_event' + html, {
>     ID: 'test'
> }, Scratch.vm.runtime.getSpriteTargetByName(util.thread.target.getName()));
> ```
> 
> it is not running the hats with the id argument set to test it just says s[e] is undefined please hlep

that option only works with fields

**Reply by RedMan13** [reply]
it throwing an error is strange but kust to note cause tou said argumwnr which sounds like input

**Reply by dumorando** [reply]
this?
![image](https://github.com/PenguinMod/PenguinMod-Home/assets/144301348/07f8f58b-afb5-415b-a792-35fd2b263003)

**Reply by JeremyGamer13** [reply]
yea startHats only works with menus, and those menus cant accept reporters

**Reply by dumorando** [reply]
ohhhhhhhhhh i see
is there any way i could make the way im doing work tho

**Reply by JeremyGamer13** [reply, marked as answer]
you basically just need to always run start hats and have the hat itself decide if it should run
this might help: https://docs.turbowarp.org/development/extensions/hats#predicate-based-hat-blocks

**Reply by JeremyGamer13** [reply]
not sure if this is supported in PM or any mods yet btw since i dont remember this being in the docs

**Reply by dumorando** [reply]
pretty sure i did something similar to this in vanilla scratch and it worked so i dont see why it wouldnt here

**Reply by RedMan13** [reply]
> not sure if this is supported in PM or any mods yet btw since i dont remember this being in the docs

it has been here since the scratch extension format was created :trollface:

**Reply by dumorando** [reply]
turns out i dident need to do that, i can just do vm.runtime.startHats("johnMyExtension_logToConsole", {TEXT:"test"}) and it works.for some reason text and number values turn into "TEXT" and idk what the number one is but it works

**Reply by dumorando** [reply]
nevermind it dosent work im gonna remake the entire scratch vm but it will be better because like i made it

**Comment by JeremyGamer13** [comment]
if the menu accepts reporters it wont work
(also whats the purpose of `Scratch.vm.runtime.getSpriteTargetByName(util.thread.target.getName())` over just `util.target` )

**Reply by dumorando** [reply]
im stupid as shit

**Reply by JeremyGamer13** [reply]
:trollface:

**Reply by RedMan13** [reply]
> im stupid as shit

hurhuhruhrurhurhrurhurhruhrrhurhrurhurhru
